packages/components/src/navigation/menu/menu-title.js

```diff
@@ -52,9 +52,8 @@ export default function NavigationMenuTitle( {
 		<MenuTitleUI className="components-navigation__menu-title">
 			{ ! isSearching && (
 				<MenuTitleHeadingUI
-					as="h2"
```

Comment:
I believe you can still use the `as` prop here if you want to keep it semantically an `h2`, but styled as a level 3. (Sorry if it doesn't work, I only read the source code)